File: icore/src/main/java/com/ddu/icore/bezier/QuadraticPointFTypeEvaluator.java
package com.ddu.icore.bezier;

import android.animation.TypeEvaluator;
import android.graphics.PointF;

/**
 * Created by yzbzz on 2019-08-05.
 */
public class QuadraticPointFTypeEvaluator implements TypeEvaluator<PointF> {

    private PointF mControl;

    public QuadraticPointFTypeEvaluator(PointF control) {
        this.mControl = control;
    }

    @Override
    public PointF evaluate(float fraction, PointF startValue, PointF endValue) {
        return BezierUtils.CalculateBezierPointForQuadratic(fraction, startValue, mControl, endValue);
    }
}

File: icore/src/main/java/com/ddu/icore/bezier/QuadraticPointFTypeEvaluator.kt
package com.ddu.icore.bezier

import android.animation.TypeEvaluator
import android.graphics.PointF

/**
 * Created by yzbzz on 2019-08-05.
 */
class QuadraticPointFTypeEvaluator(private val mControl: PointF) : TypeEvaluator<PointF> {

    override fun evaluate(fraction: Float, startValue: PointF, endValue: PointF): PointF {
        return BezierUtils.calculateBezierPointForQuadratic(fraction, startValue, mControl, endValue)
    }
}
